### PR TITLE
Use Sass for CSS, Visual Refresh

### PR DESCRIPTION
**Pull request by hramos**
This is the first step towards cleaning up the site's CSS, with the goal of eventually grouping together classes common to the various FB OSS websites built from the same template.

Also added in this PR is a slight website refresh.

- New docs sidebar
- Lighter weight headers
- Consistent use of colors across the site

## Test Plan

Tested on Chrome, Firefox, iPhone, iPad.

screen shot 2017-02-08 at 5 42 43 pm

screen shot 2017-02-08 at 5 43 12 pm

### CODE REVIEW

File: website/src/react-native/css/react-native.css [hunk 1]
```diff
@@ -1,532 +1,778 @@
+/*! normalize.css v5.0.0 | MIT License | github.com/necolas/normalize.css */
```
Comment (hramos) [inline website/src/react-native/css/react-native.css]: This entire file is autogenerated by Sass. No need to review it, other than to confirm all partials/imports are handled successfully during site generation.

Comment (ericvicenti) [inline website/src/react-native/css/react-native.css]: Does it need to be checked in to the repo? Why can't the CI generate this?

**Comment by hramos** [comment]
@ericvicenti unfortunately it does need to be checked in. If `src/react-native/css/react-native.css` is added to `website/.gitignore`, then the `publish-gh-pages.js` script will not commit the generated file to the `gh-pages` branch during deploy.

**Comment by hramos** [comment]
Closing in favor of #12575

**Pull request closed without merging**